pkg/broker/filter/filter_handler.go

```diff
@@ -246,24 +288,47 @@ func (h *Handler) sendEvent(ctx context.Context, headers http.Header, target str
 
 	err = kncloudevents.WriteHTTPRequestWithAdditionalHeaders(ctx, message, req, additionalHeaders)
 	if err != nil {
-		return nil, fmt.Errorf("failed to write request: %w", err)
+		responseErr.err = fmt.Errorf("failed to write request: %w", err)
+		return nil, responseErr
 	}
 
 	start := time.Now()
 	resp, err := h.sender.Send(req)
 	dispatchTime := time.Since(start)
 	if err != nil {
-		err = fmt.Errorf("failed to dispatch message: %w", err)
+		responseErr.ResponseCode = http.StatusInternalServerError
+		responseErr.ResponseBody = []byte(fmt.Sprintf("dispatch error: %s", err.Error()))
+		responseErr.err = fmt.Errorf("failed to dispatch message: %w", err)
+		return resp, responseErr
 	}
 
 	sc := 0
 	if resp != nil {
 		sc = resp.StatusCode
+		responseErr.ResponseCode = sc
 	}
 
 	_ = h.reporter.ReportEventDispatchTime(reporterArgs, sc, dispatchTime)
 
-	return resp, err
+	if resp.StatusCode < http.StatusOK ||
```

Comment:
@pierDipi  Because in there, we add some logic to handle the requests(responseCode != 2xx), we read io.Reader in this logic: https://stackoverflow.com/questions/30910487/why-does-an-io-reader-become-empty-after-reading-it, so when responseCode is 429 and the request has additionalReplyHeaders, it could not get the expectedResponseHeaders, so I removed these two lines.